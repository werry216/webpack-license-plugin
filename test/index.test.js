const compileWebpack = require('./testUtils').compileWebpack
const emptyOptions = require('../example/empty/webpack.config')
const basicOptions = require('../example/basic/webpack.config')
const blacklistOptions = require('../example/blacklisted-licenses/webpack.config')
const fs = require('fs')

jest.setTimeout(30000)

describe('WebpackLicensePlugin', () => {
  test("empty example doesn't emit a file", async () => {
    const stats = await compileWebpack(emptyOptions)
    const files = stats.toJson().assets.map(x => x.name)

    expect(files.includes('oss-licenses.json')).toBe(false)
  })

  test('basic example emits a file with the right contents', async () => {
    const stats = await compileWebpack(basicOptions)
    const files = stats.toJson().assets.map(x => x.name)

    expect(files.includes('oss-licenses.json')).toBe(true)
    const file = fs.readFileSync('./example/basic/public/oss-licenses.json').toString()
    expect(file).toMatchSnapshot()
  })

  test('blacklisted-licenses example throws', async () => {
    const compile = async () => await compileWebpack(blacklistOptions)
    await expect(compile()).rejects.toThrow(/WebpackLicensePlugin: found blacklisted license "MIT"/)
  })
})
