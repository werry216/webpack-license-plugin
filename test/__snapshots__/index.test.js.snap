// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebpackLicensePlugin basic example emits a file with the right contents 1`] = `
"{
  \\"summary\\": {
    \\"MIT\\": 4
  },
  \\"packages\\": [
    {
      \\"name\\": \\"fbjs\\",
      \\"version\\": \\"0.8.17\\",
      \\"repository\\": \\"https://github.com/facebook/fbjs\\",
      \\"source\\": \\"https://registry.npmjs.org/fbjs/-/fbjs-0.8.17.tgz\\",
      \\"license\\": \\"MIT\\",
      \\"licenseText\\": \\"MIT License\\\\n\\\\nCopyright (c) 2013-present, Facebook, Inc.\\\\n\\\\nPermission is hereby granted, free of charge, to any person obtaining a copy of\\\\nthis software and associated documentation files (the \\\\\\"Software\\\\\\"), to deal in\\\\nthe Software without restriction, including without limitation the rights to\\\\nuse, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of\\\\nthe Software, and to permit persons to whom the Software is furnished to do so,\\\\nsubject to the following conditions:\\\\n\\\\nThe above copyright notice and this permission notice shall be included in all\\\\ncopies or substantial portions of the Software.\\\\n\\\\nTHE SOFTWARE IS PROVIDED \\\\\\"AS IS\\\\\\", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR\\\\nIMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS\\\\nFOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR\\\\nCOPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER\\\\nIN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN\\\\nCONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.\\\\n\\"
    },
    {
      \\"name\\": \\"object-assign\\",
      \\"version\\": \\"4.1.1\\",
      \\"author\\": \\"Sindre Sorhus\\",
      \\"repository\\": \\"https://github.com/sindresorhus/object-assign\\",
      \\"source\\": \\"https://registry.npmjs.org/object-assign/-/object-assign-4.1.1.tgz\\",
      \\"license\\": \\"MIT\\",
      \\"licenseText\\": \\"The MIT License (MIT)\\\\n\\\\nCopyright (c) Sindre Sorhus <sindresorhus@gmail.com> (sindresorhus.com)\\\\n\\\\nPermission is hereby granted, free of charge, to any person obtaining a copy\\\\nof this software and associated documentation files (the \\\\\\"Software\\\\\\"), to deal\\\\nin the Software without restriction, including without limitation the rights\\\\nto use, copy, modify, merge, publish, distribute, sublicense, and/or sell\\\\ncopies of the Software, and to permit persons to whom the Software is\\\\nfurnished to do so, subject to the following conditions:\\\\n\\\\nThe above copyright notice and this permission notice shall be included in\\\\nall copies or substantial portions of the Software.\\\\n\\\\nTHE SOFTWARE IS PROVIDED \\\\\\"AS IS\\\\\\", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR\\\\nIMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,\\\\nFITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE\\\\nAUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER\\\\nLIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,\\\\nOUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN\\\\nTHE SOFTWARE.\\\\n\\"
    },
    {
      \\"name\\": \\"react-dom\\",
      \\"version\\": \\"16.4.2\\",
      \\"repository\\": \\"https://github.com/facebook/react\\",
      \\"source\\": \\"https://registry.npmjs.org/react-dom/-/react-dom-16.4.2.tgz\\",
      \\"license\\": \\"MIT\\",
      \\"licenseText\\": \\"MIT License\\\\n\\\\nCopyright (c) 2013-present, Facebook, Inc.\\\\n\\\\nPermission is hereby granted, free of charge, to any person obtaining a copy\\\\nof this software and associated documentation files (the \\\\\\"Software\\\\\\"), to deal\\\\nin the Software without restriction, including without limitation the rights\\\\nto use, copy, modify, merge, publish, distribute, sublicense, and/or sell\\\\ncopies of the Software, and to permit persons to whom the Software is\\\\nfurnished to do so, subject to the following conditions:\\\\n\\\\nThe above copyright notice and this permission notice shall be included in all\\\\ncopies or substantial portions of the Software.\\\\n\\\\nTHE SOFTWARE IS PROVIDED \\\\\\"AS IS\\\\\\", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR\\\\nIMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,\\\\nFITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE\\\\nAUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER\\\\nLIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,\\\\nOUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE\\\\nSOFTWARE.\\\\n\\"
    },
    {
      \\"name\\": \\"react\\",
      \\"version\\": \\"16.4.2\\",
      \\"repository\\": \\"https://github.com/facebook/react\\",
      \\"source\\": \\"https://registry.npmjs.org/react/-/react-16.4.2.tgz\\",
      \\"license\\": \\"MIT\\",
      \\"licenseText\\": \\"MIT License\\\\n\\\\nCopyright (c) 2013-present, Facebook, Inc.\\\\n\\\\nPermission is hereby granted, free of charge, to any person obtaining a copy\\\\nof this software and associated documentation files (the \\\\\\"Software\\\\\\"), to deal\\\\nin the Software without restriction, including without limitation the rights\\\\nto use, copy, modify, merge, publish, distribute, sublicense, and/or sell\\\\ncopies of the Software, and to permit persons to whom the Software is\\\\nfurnished to do so, subject to the following conditions:\\\\n\\\\nThe above copyright notice and this permission notice shall be included in all\\\\ncopies or substantial portions of the Software.\\\\n\\\\nTHE SOFTWARE IS PROVIDED \\\\\\"AS IS\\\\\\", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR\\\\nIMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,\\\\nFITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE\\\\nAUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER\\\\nLIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,\\\\nOUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE\\\\nSOFTWARE.\\\\n\\"
    }
  ]
}"
`;

exports[`WebpackLicensePlugin packages with unknown licenseFiles work 1`] = `
"{
  \\"summary\\": {
    \\"MIT\\": 4
  },
  \\"packages\\": [
    {
      \\"name\\": \\"fbjs\\",
      \\"version\\": \\"0.8.17\\",
      \\"repository\\": \\"https://github.com/facebook/fbjs\\",
      \\"source\\": \\"https://registry.npmjs.org/fbjs/-/fbjs-0.8.17.tgz\\",
      \\"license\\": \\"MIT\\",
      \\"licenseText\\": \\"MIT License\\\\n\\\\nCopyright (c) 2013-present, Facebook, Inc.\\\\n\\\\nPermission is hereby granted, free of charge, to any person obtaining a copy of\\\\nthis software and associated documentation files (the \\\\\\"Software\\\\\\"), to deal in\\\\nthe Software without restriction, including without limitation the rights to\\\\nuse, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of\\\\nthe Software, and to permit persons to whom the Software is furnished to do so,\\\\nsubject to the following conditions:\\\\n\\\\nThe above copyright notice and this permission notice shall be included in all\\\\ncopies or substantial portions of the Software.\\\\n\\\\nTHE SOFTWARE IS PROVIDED \\\\\\"AS IS\\\\\\", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR\\\\nIMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS\\\\nFOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR\\\\nCOPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER\\\\nIN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN\\\\nCONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.\\\\n\\"
    },
    {
      \\"name\\": \\"object-assign\\",
      \\"version\\": \\"4.1.1\\",
      \\"author\\": \\"Sindre Sorhus\\",
      \\"repository\\": \\"https://github.com/sindresorhus/object-assign\\",
      \\"source\\": \\"https://registry.npmjs.org/object-assign/-/object-assign-4.1.1.tgz\\",
      \\"license\\": \\"MIT\\",
      \\"licenseText\\": \\"The MIT License (MIT)\\\\n\\\\nCopyright (c) Sindre Sorhus <sindresorhus@gmail.com> (sindresorhus.com)\\\\n\\\\nPermission is hereby granted, free of charge, to any person obtaining a copy\\\\nof this software and associated documentation files (the \\\\\\"Software\\\\\\"), to deal\\\\nin the Software without restriction, including without limitation the rights\\\\nto use, copy, modify, merge, publish, distribute, sublicense, and/or sell\\\\ncopies of the Software, and to permit persons to whom the Software is\\\\nfurnished to do so, subject to the following conditions:\\\\n\\\\nThe above copyright notice and this permission notice shall be included in\\\\nall copies or substantial portions of the Software.\\\\n\\\\nTHE SOFTWARE IS PROVIDED \\\\\\"AS IS\\\\\\", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR\\\\nIMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,\\\\nFITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE\\\\nAUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER\\\\nLIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,\\\\nOUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN\\\\nTHE SOFTWARE.\\\\n\\"
    },
    {
      \\"name\\": \\"react-dom\\",
      \\"version\\": \\"16.4.2\\",
      \\"repository\\": \\"https://github.com/facebook/react\\",
      \\"source\\": \\"https://registry.npmjs.org/react-dom/-/react-dom-16.4.2.tgz\\",
      \\"license\\": \\"MIT\\",
      \\"licenseText\\": \\"MIT License\\\\n\\\\nCopyright (c) 2013-present, Facebook, Inc.\\\\n\\\\nPermission is hereby granted, free of charge, to any person obtaining a copy\\\\nof this software and associated documentation files (the \\\\\\"Software\\\\\\"), to deal\\\\nin the Software without restriction, including without limitation the rights\\\\nto use, copy, modify, merge, publish, distribute, sublicense, and/or sell\\\\ncopies of the Software, and to permit persons to whom the Software is\\\\nfurnished to do so, subject to the following conditions:\\\\n\\\\nThe above copyright notice and this permission notice shall be included in all\\\\ncopies or substantial portions of the Software.\\\\n\\\\nTHE SOFTWARE IS PROVIDED \\\\\\"AS IS\\\\\\", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR\\\\nIMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,\\\\nFITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE\\\\nAUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER\\\\nLIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,\\\\nOUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE\\\\nSOFTWARE.\\\\n\\"
    },
    {
      \\"name\\": \\"react\\",
      \\"version\\": \\"16.4.2\\",
      \\"repository\\": \\"https://github.com/facebook/react\\",
      \\"source\\": \\"https://registry.npmjs.org/react/-/react-16.4.2.tgz\\",
      \\"license\\": \\"MIT\\",
      \\"licenseText\\": \\"MIT License\\\\n\\\\nCopyright (c) 2013-present, Facebook, Inc.\\\\n\\\\nPermission is hereby granted, free of charge, to any person obtaining a copy\\\\nof this software and associated documentation files (the \\\\\\"Software\\\\\\"), to deal\\\\nin the Software without restriction, including without limitation the rights\\\\nto use, copy, modify, merge, publish, distribute, sublicense, and/or sell\\\\ncopies of the Software, and to permit persons to whom the Software is\\\\nfurnished to do so, subject to the following conditions:\\\\n\\\\nThe above copyright notice and this permission notice shall be included in all\\\\ncopies or substantial portions of the Software.\\\\n\\\\nTHE SOFTWARE IS PROVIDED \\\\\\"AS IS\\\\\\", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR\\\\nIMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,\\\\nFITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE\\\\nAUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER\\\\nLIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,\\\\nOUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE\\\\nSOFTWARE.\\\\n\\"
    }
  ]
}"
`;
